plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -436,61 +501,226 @@ fu_nordic_hid_cfg_channel_add_peers(FuNordicHidCfgChannel *self, GError **error)
 						0,
 						error))
 		return FALSE;
+
 	if (fu_nordic_hid_cfg_channel_cmd_receive(self,
 						  CONFIG_STATUS_DISCONNECTED,
 						  res,
 						  &error_local)) {
-		/* no peers */
+		/* forwarding configuration channel to peers not supported */
 		return TRUE;
 	}
 
 	/* Peers available */
 	if (!fu_nordic_hid_cfg_channel_cmd_receive(self, CONFIG_STATUS_SUCCESS, res, error))
 		return FALSE;
 
+	*cmd_supported = TRUE;
+	return TRUE;
+}
+
+static gboolean
+fu_nordic_hid_cfg_channel_get_next_peer_id_cmd(FuNordicHidCfgChannel *self,
+					       GError **error,
+					       guint8 *peer_id)
+{
+	g_autoptr(FuNordicCfgChannelMsg) res = g_new0(FuNordicCfgChannelMsg, 1);
+
+	if (!fu_nordic_hid_cfg_channel_cmd_send(self,
+						NULL,
+						NULL,
+						CONFIG_STATUS_GET_PEER,
+						NULL,
+						0,
+						error))
+		return FALSE;
+
+	if (!fu_nordic_hid_cfg_channel_cmd_receive(self, CONFIG_STATUS_SUCCESS, res, error))
+		return FALSE;
+
+	*peer_id = res->data[8];
+
+	return TRUE;
+}
+
+static gboolean
+fu_nordic_hid_cfg_channel_read_peers_cache_cmd(FuNordicHidCfgChannel *self,
+					       GError **error,
+					       gboolean *cmd_supported,
+					       guint8 peers_cache[PEERS_CACHE_LEN])
+{
+	g_autoptr(FuNordicCfgChannelMsg) res = g_new0(FuNordicCfgChannelMsg, 1);
+	g_autoptr(GError) error_local = NULL;
+
+	*cmd_supported = FALSE;
+
+	if (!fu_nordic_hid_cfg_channel_cmd_send(self,
+						NULL,
+						NULL,
+						CONFIG_STATUS_GET_PEERS_CACHE,
+						NULL,
+						0,
+						error))
+		return FALSE;
+
+	if (fu_nordic_hid_cfg_channel_cmd_receive(self,
+						  CONFIG_STATUS_DISCONNECTED,
+						  res,
+						  &error_local)) {
+		/* Configuration channel peers cache not supported. */
```

Comment:
configuration

Reply:
i think that there should not be also period at the end, same in lines: 609, 624, 657, 682, 716